internal/users/manager.go

```diff
@@ -212,6 +226,42 @@ func (m *Manager) UpdateUser(u types.UserInfo) (err error) {
 	return nil
 }
 
+// checkGroupNameConflict checks if a group with the given name already exists.
+// If it does, it checks if it has the same UGID.
+func (m *Manager) checkGroupNameConflict(name string, ugid string) error {
+	// Check if a group with that name already exists. First check in our database.
+	existingGroup, err := m.cache.GroupByName(name)
+	if err != nil && !errors.Is(err, cache.NoDataFoundError{}) {
+		// Unexpected error
+		return err
+	}
+
+	if errors.Is(err, cache.NoDataFoundError{}) {
+		// The group does not exist in the database, check if it exists on the system.
+		existingGroup, err := user.LookupGroup(name)
+		var unknownGroupErr user.UnknownGroupError
+		if !errors.As(err, &unknownGroupErr) {
```

Comment:
Can't you use just `errors.Is` since you're not using `unknownGroupErr`?

Reply:
No, `errors.Is` checks if two error *values* are the same. Since the `UnknownGroupError` contains the group name, we have to use `errors.As` to check if it has the `UnknownGroupError` type.

You can use `errors.Is` when the error is defined like this:

```go
ErrNotExist   = errors.New("file does not exist")
```

You can't use it when the error is defined like this:

```
type UnknownGroupError string

func (e UnknownGroupError) Error() string {
	return "group: unknown group " + string(e)
}
```

Reply:
Ah, right... Annoying error definition...